test/e2e/validation_test.go

```diff
@@ -373,33 +379,36 @@ func (tc *testContext) runJob(name string, command []string) (string, error) {
 	return logs, nil
 }
 
-// getWinServices returns a map of Windows services from the instance with the given address, the key:value format being
-// name:status
-func (tc *testContext) getWinServices(addr string) (map[string]string, error) {
-	// This command returns CR+newline separated quoted CSV entries consisting of service name and status. For example:
-	// "kubelet","Running"\r\n"VaultSvc","Stopped"
-	command := "Get-Service | Select-Object -Property Name,Status | ConvertTo-Csv -NoTypeInformation"
+// getWinServices returns a map of Windows services from the instance with the given address, the map key being the
+// service's name
+func (tc *testContext) getWinServices(addr string) (map[string]winService, error) {
+	// This command returns CR+newline separated quoted CSV entries consisting of service name, state and description.
+	// For example: "kubelet","Running","OpenShift managed kubelet"\r\n"VaultSvc","Stopped",
+	command := "Get-CimInstance -ClassName Win32_Service | Select-Object -Property Name,State,Description | " +
+		"ConvertTo-Csv -NoTypeInformation"
 	out, err := tc.runPowerShellSSHJob("get-windows-svc-list", command, addr)
 	if err != nil {
 		return nil, errors.Wrap(err, "error running SSH job")
 	}
 
 	// Remove the header and trailing whitespace from the command output
-	outSplit := strings.SplitAfterN(out, "\"Name\",\"Status\"\r\n", 2)
+	outSplit := strings.SplitAfterN(out, "\"Name\",\"State\",\"Description\"\r\n", 2)
 	if len(outSplit) != 2 {
 		return nil, errors.New("unexpected command output: " + out)
 	}
 	trimmedList := strings.TrimSpace(outSplit[1])
 
 	// Make a map from the services, removing the quotes around each entry
-	services := make(map[string]string)
+	services := make(map[string]winService)
 	lines := strings.Split(trimmedList, "\r\n")
 	for _, line := range lines {
-		fields := strings.Split(line, ",")
-		if len(fields) != 2 {
+		// Split into 3 substrings, Name, State, Description. The description can contain a comma, so SplitN is required
+		fields := strings.SplitN(line, ",", 3)
+		if len(fields) != 3 {
 			return nil, errors.New("expected comma separated values, found: " + line)
 		}
-		services[strings.Trim(fields[0], "\"")] = strings.Trim(fields[1], "\"")
+		services[strings.Trim(fields[0], "\"")] = winService{state: strings.Trim(fields[1], "\""),
+			description: strings.Trim(fields[2], "\"")}
```

Comment:
To improve readability,  consider doing the cleanup first, then the assignation.
For example:
```
name := strings.Trim(fields[0], "\"")
state := strings.Trim(fields[1], "\"")
...
services[name] = winService{state: state, ... }

```

Reply:
While it applies in this case, please refrain from adding single use variables in general for the sake of "readability".